### ISSUE TITLE
need to be blank for [search][value] after remove the filter

### ISSUE THREAD

**Issue by BAndy18ru** [vedmack/yadcf#328]
Hi, great job!

I just found this and simple tested)
Filters not removed properly after X button.
It happen on Date and Range of numbers filters, "-yadcf_delim-" sended to the server

Also, can you pls add to the code minimized version of JS.

Thanks

**Comment by vedmack** [comment]
If you'll provide a link to a test case (using my local[ yadcf copy](https://dl.dropboxusercontent.com/u/55621125/yadcf/war/resources/js/jquery.dataTables.yadcf.js)) I will be able to fix it much quicker

**Comment by louking** [comment]
I see similar in http://yadcf-showcase.appspot.com/server_side_source.html where on initial load `columns[4][search][value]` is empty, but after moving slider, then clearing with X `columns[4][search][value]` becomes `0-yadcf_delim-134`. Of course, server side can be built to handle `-yadcf_delim-` equivalent to empty value, equivalent to full range...

**Comment by louking** [comment]
Just determined the issue originally reported by @BAndy18ru happens with a date_range filter, and can be reproduced in  http://yadcf-showcase.appspot.com/server_side_source.html

**Comment by true-gler** [comment]
same here in select2 

after removing entries the input field (with [x] or with backspace) is not cleared properly (or connection in regexp remains)
Got error 'empty (sub)expression' from regexp 
```php
LOWER(`table`.`field`) REGEXP |filterString)
```

after 
``` php
yadcf.doFilter('clear', '-table', 24); 
```

the input is cleared and its working again

**Comment by vedmack** [comment]
@louking , can you please write down a simple flow with expected / actual result so I can understand whats not working as expected? I tried to reset the range date with [X], and it sends `-yadcf_delim-` to server which is fine, so please elaborate

**Comment by louking** [comment]
Follow these steps with http://yadcf-showcase.appspot.com/server_side_source.html

1. clear `Engine` filter with X (just to see values for `Just a Date` and `Just a Number`)

`columns[3][search][value]` is empty
`columns[4][search][value]` is empty

2. pick a date in `Just a Date`: `To`

`columns[3][search][value]: -yadcf_delim-12/13/2016`

3. clear `Just a Date` filter with X

`columns[3][search][value]: -yadcf_delim-` - this may be valid but I think most people would expect this to be empty as the state is the same as after item 1

4. pick a number for `Just a Number` `To`

`columns[4][search][value]: 0-yadcf_delim-131`

5. clear `Just a Number` filter with X

`columns[4][search][value]: 0-yadcf_delim-134` - this is is interesting as why isn't this empty or `-yadcf-delim-`?

Frankly, I don't have a problem with this behavior and have coded around it. Having said that, I think it would be more intuitive when the filter is cleared to have the search value go back to the same state as it was before it was ever used (empty)

**Comment by vedmack** [comment]
fixed in [0.9.2.beta.4](https://github.com/vedmack/yadcf/blob/eac783b7a567db08ab8a9703d1887daa34faadc9/src/jquery.dataTables.yadcf.js)
